From: cail <jeff@forristal.com>
Date: Tue, 22 Dec 2020 21:09:45 +0000
Subject: report 48M not 46.5M

---
 core/rtw_recv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/rtw_recv.c b/core/rtw_recv.c
index 446d2c5..27e3595 100644
--- a/core/rtw_recv.c
+++ b/core/rtw_recv.c
@@ -3882,7 +3882,7 @@ static sint fill_radiotap_hdr(_adapter *padapter, union recv_frame *precvframe,
 
 	u8 data_rate[] = {
 		2, 4, 11, 22, /* CCK */
-		12, 18, 24, 36, 48, 72, 93, 108, /* OFDM */
+		12, 18, 24, 36, 48, 72, 96, 108, /* OFDM */
 		0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, /* HT MCS index */
 		16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31,
 		0, 1, 2, 3, 4, 5, 6, 7, 8, 9, /* VHT Nss 1 */
